theSrc/scripts/PlotlyChartElements.js

```diff
@@ -0,0 +1,253 @@
+            dtick: parseTickDistance(config.yBoundsUnitsMajor),
+            tickprefix: config.yPrefix,
+            ticksuffix: config.ySuffix,
+            automargin: true,
+            layer: 'below traces'
+        },
+        title: {
+            text: config.title,
+            font: {
+                family: config.titleFontFamily,
+                color: config.titleFontColor,
+                size: config.titleFontSize
+            },
+            xref: 'paper',
+            automargin: false // setting this to true stuffs up alignment
+        },
+        showlegend: config.legendShow,
+        legend: {
+            font: {
+                family: config.legendFontFamily,
+                color: config.legendFontColor,
+                size: config.legendFontSize
+            },
+            itemsizing: 'constant',
+            yref: 'paper',
+            y: 0.5,
+            yanchor: 'middle',
+        },
+        margin: {
+            t: 20 + (config.title.length ? config.titleFontSize * 1.25 : 0),
+            b: 20,
+            r: 60,
+            l: 80,
+            automargin: true
+        },
+        hoverlabel: {
+            namelength: -1, // prevents trace name truncating
+            bordercolor: 'transparent',
+            font: {
+                family: config.tooltipFontFamily,
+                size: config.tooltipFontSize
+            }
+        },
+        shapes: addLines(config),
+    }
+    return plot_layout
+}
+
+function addLines (config) {
+    const lines = []
+    if (config.origin) {
+        lines.push({
+            type: 'line',
+            layer: 'above',
+            line: { color: '#000', width: 1, dash: 'dot' },
+            x0: 0,
+            x1: 0,
+            xref: 'x',
+            y0: 0,
+            y1: 1,
+            yref: 'paper'
+        })
+        lines.push({
+            type: 'line',
+            layer: 'above',
+            line: { color: '#000', width: 1, dash: 'dot' },
+            y0: 0,
+            y1: 0,
+            yref: 'y',
+            x0: 0,
+            x1: 1,
+            xref: 'paper'
+        })
+    }
+    if (config.plotBorderShow) {
+        lines.push({
+            type: 'line',
+            layer: 'below',
+            line: { color: config.plotBorderColor, width: config.plotBorderWidth },
+            y0: 1,
+            y1: 1,
+            yref: 'paper',
+            x0: 0,
+            x1: 1,
+            xref: 'paper'
+        })
+        lines.push({
+            type: 'line',
+            layer: 'below',
+            line: { color: config.plotBorderColor, width: config.plotBorderWidth },
+            y0: 0,
+            y1: 1,
+            yref: 'paper',
+            x0: 1,
+            x1: 1,
+            xref: 'paper'
+        })
+    }
+    return lines
+}
+
+function parseTickDistance (x) {
+    if (x === undefined) return null
+    return x
+}
+
+function autoFontColor (bg_color) {
```

Comment:
Is this from R?

Reply:
Maybe it could have a better name?

Reply:
Yes, this was a function in flipStandardCharts. I've renamed it to `blackOrWhite()`